### PR TITLE
Adding Pagination

### PR DESCRIPTION
**Pull request by betrevisan**
The proposed commits in this PR implement a pagination feature that allows the user to move through comments using different pages when not all comments can fit into a single page. Changes include adding the pagination navigation bar, writing the getPagination function that is triggered when contact.html load, creating the pagination servlet, writing its doGet and doPost functions, and writing the necessary javascript code to put the pagination buttons together. In order to implement this feature, I realized it would be helpful to create more object classes and more servlets. With that in mind, I created a new object class called AllComments that keeps track of the information regarding all the comments in the datastore.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/AllComments.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  private long total;
+  private long max;
+  private long page;
+
+  public AllComments(long total, long max, long page) {
+    this.total = total;
+    this.max = max;
+    this.page = page;
+  }
+}
```
Comment (kashiish) [inline portfolio/src/main/java/com/google/sps/data/AllComments.java:29]: Should add a new line here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -30,127 +30,217 @@
...
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Query query = new Query("AllComments");
+        PreparedQuery resultsAll = datastore.prepare(query);
+        Iterator<Entity> allIter = resultsAll.asIterator();
+        Entity allComments = allIter.next();
+        long page = (long) allComments.getProperty("page");
+
+        for (int count = 0; count < (maxComments * page) && count < totalComments; count++) {
+            Entity entity = iter.next();
+            if (count >= (maxComments * (page - 1))) {
```
Comment (kashiish) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:172]: Maybe add a couple comments here, it took a while for me to understand exactly what was happening here.

**Comment by betrevisan** [comment]
> Just left a few comments, but this is awesome! Glad you figured it out.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (betrevisan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Yeah it does look confusing. Thanks!

**Pull request merged**